library/src/containers/Servers/ServerSecurity.tsx

```diff
@@ -8,82 +8,165 @@ import { ServerHelpers } from '../../helpers';
 
 interface Props {
   serverRequirements: ServerSecurityRequirement[];
+  protocol: string;
 }
 
 export const ServerSecurity: React.FunctionComponent<Props> = ({
   serverRequirements = [],
+  protocol = '',
 }) => {
   const asyncapi = useSpec();
   const securitySchemes =
     asyncapi.hasComponents() && asyncapi.components().securitySchemes();
 
+  let renderedServerSecurities;
   if (
+    !serverRequirements ||
     !serverRequirements.length ||
     !securitySchemes ||
     !Object.keys(securitySchemes).length
   ) {
-    return null;
-  }
-
-  const serverSecurities: React.ReactNodeArray = serverRequirements
-    .map(requirement => {
-      const def: SecurityScheme =
-        securitySchemes[Object.keys(requirement.json())[0]];
-
-      if (!def) {
-        return null;
-      }
-      return <ServerSecurityItem securitySchema={def} key={def.type()} />;
-    })
-    .filter(Boolean);
-
-  if (!serverSecurities.length) {
-    return null;
-  }
+    if (protocol === 'kafka' || protocol === 'kafka-secure') {
+      renderedServerSecurities = (
+        <ServerSecurityItem protocol={protocol} securitySchema={null} key="" />
+      );
+    }
+  } else {
+    const serverSecurities: React.ReactNodeArray = serverRequirements
+      .map(requirement => {
+        const def: SecurityScheme =
+          securitySchemes[Object.keys(requirement.json())[0]];
 
-  return (
-    <div className="text-sm mt-4">
-      <h5 className="text-gray-700 text-base">Security:</h5>
+        if (!def) {
+          return null;
+        }
+        return (
+          <ServerSecurityItem
+            protocol={protocol}
+            securitySchema={def}
+            key={def.type()}
+          />
+        );
+      })
+      .filter(Boolean);
+    renderedServerSecurities = (
       <ul>
         {serverSecurities.map((security, idx) => (
           <li className="mt-2" key={idx}>
             {security}
           </li>
         ))}
       </ul>
+    );
+  }
+
+  return (
+    <div className="text-sm mt-4">
+      {renderedServerSecurities && (
+        <h5 className="text-gray-700 text-base">Security:</h5>
+      )}
+      {renderedServerSecurities}
     </div>
   );
 };
 
 interface ServerSecurityItemProps {
-  securitySchema: SecurityScheme;
+  securitySchema: SecurityScheme | null;
+  protocol: string;
 }
 
 const ServerSecurityItem: React.FunctionComponent<ServerSecurityItemProps> = ({
   securitySchema,
+  protocol,
 }) => {
   const schemas: React.ReactNodeArray = [];
-  if (securitySchema.name()) {
+  if (securitySchema && securitySchema.name()) {
     schemas.push(<span>Name: {securitySchema.name()}</span>);
   }
-  if (securitySchema.in()) {
+  if (securitySchema && securitySchema.in()) {
```

Comment:
Instead of checking the `securitySchema` value in every `if`, you can add only one `if` and inside them another, like:

```js
if (securitySchema) {
  if (securitySchema.name()) { ... }
  ...
}
```

But it's not required by me, just a nice-to-have suggestion.